components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -197,6 +206,13 @@ - (void)dealloc {
 - (CGFloat)dynamicBottomMargin {
   CGFloat keyboardHeight = self.watcher.visibleKeyboardHeight;
   CGFloat userHeight = self.bottomOffset;
+  if (!_snackbarView.message.usesLegacySnackbar) {
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+    if (@available(iOS 11.0, *)) {
+      userHeight = MAX(userHeight, self.window.safeAreaInsets.bottom);
```

Comment:
Just to confirm: snackbar views are always presented on the window?

Or put another way, is it possible for a snackbar to be presented as part of a smaller-than-the-window view controller (e.g. on iPad)?

If it's always on the window then this code is safe, otherwise we'll have to read the safe area insets from the container view.

Reply:
Note that it may be safer to try to make use of the container view's safe area insets anyway.

Reply:
done